### PR TITLE
fix field name flash

### PR DESCRIPTION
**Pull request by jjidt**
- adds a nifty usePrevious hook that will return the previous value on each re-render
- adds a check for generic name changes and image name changes, if the incoming field.name is the same as the previous one, we know not to update the current.value

### CODE REVIEW

**Review by dheniges — CHANGES_REQUESTED** [review]
I'm still getting the flash on this, it seems like there's a point where the previousValue matches the field name in redux here after a render but before the PUT response is received

**Comment by jjidt** [comment]
added an isUpdating check there as well and it seems to behave better when wildly clicking around and updating things. This probably won't completely solve it for super slow connections, so we might want to implement some kind of component update date vs. lastUpdatedAt date check in the future, possibly?

**Review by dheniges — COMMENTED** [review]
I think the isUpdating check fixes the issue. I removed the previousField code and it still works as expected, so I don't think we want/need that previousField check here?

File: web/src/client/js/components/DocumentFields/DocumentFieldComponent.js [hunk 1]
```diff
@@ -83,12 +85,12 @@ const DocumentFieldComponent = ({
...
-    // Accept remote changes to field if it's not focused and the conflict popper isn't open
-    if (!isCurrentlyFocusedField && !showConflictPopper) {
+    // Accept remote changes to field if it's not focused, the conflict popper isn't open, and the field value has changed
+    if (!isCurrentlyFocusedField && !showConflictPopper && previousField !== field && !isUpdating) {
       setCurrentValue(field.value)
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [field.id, field.value])
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [field, isUpdating])
```
Comment (dheniges) [inline web/src/client/js/components/DocumentFields/DocumentFieldComponent.js]: I think this needs previousField as a dependency

**Review by dheniges — CHANGES_REQUESTED** [review]
This fixes my flashing problems. I think it needs one dependency update in the useEffect though

**Comment by jjidt** [comment]
updated

**Review by dheniges — APPROVED** [review]
:realjj:

**Pull request merged**